providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/grpc/EventStreamObserver.java

```diff
@@ -46,25 +55,20 @@ public void onError(Throwable t) {
         if (this.cache.getEnabled()) {
             this.cache.clear();
```

Comment:
I believe we should not bust the cache in case of error and instead use the latest known values until the stream is reconnected

Reply:
Added to https://github.com/open-feature/java-sdk-contrib/issues/390

Reply:
The problem is we don't cache all values, we only cache static flags (flags without rules).

This puts us in a situation where we really can't evaluate all flags. If we were evaluating flags in memory, we could just have "STALE" flags, but with the RPC design, we simply can't evaluate dynamic flags at all.

Reply:
This screams for a configurable option as @Kavindu-Dodan suggests 🤔 
I'll try to explain my thoughts: since values are static, I want to "remotely rewrite" the default value in the code. Hence, saving the last known values would allow me to evaluate things as the last point in time I had a connection to flagd. 
For dynamic flags, failing and defaulting to the default value is fine since I cannot pass the ctx to flagd to evaluate the flags according to the rules.